Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type-v7.md

```diff
@@ -0,0 +1,137 @@
+---
+versionFrom: 7.0.0
+---
+
+# Adding a type to the provider model
+
+To add a new type, no matter if it's a workflow, field, data source, etc, there is a number of tasks to perform to connect to the Forms provider model. This chapter walks through each step and describes how each part works. This chapter will reference the creation of a workflow type. It is, however, the same process for all types.
+
+## Preparations
+
+Create a new ASP.NET or class project in Visual Studio add references to the Umbraco.Forms.Core.dll.
+
+## Adding the type to Forms
+
+The Forms API contains a collection of classes that the provider model automatically registers. So to add a new type to Forms you inherit from the right class. In the sample below we use the class for the workflow type.
+
+```csharp
+public class Class1 : Umbraco.Forms.Core.WorkflowType
+{
+    public override WorkflowExecutionStatus Execute(Umbraco.Forms.Core.Record record)
+    {
+        throw new NotImplementedException();
+    }
+
+    public override List<Exception> ValidateSettings() {
+        throw new NotImplementedException();
+    }
+}
+```
+
+When you implement this class you get two methods added. One of them is Execute which performs the execution of the workflow and the other is a method which validates the workflow settings, we will get back to these settings later on.
+
+Even though we have the class inheritance in place, we still need to add a bit of default information.
+
+## Setting up basic type information
+
+Even though we have the class inheritance in place, we still need to add a bit of default information. This information is added in the class's empty constructor like this:
+
+```csharp
+public Class1() {
```

Comment:
Again please change the name, maybe call it `LogWorkflow` or similar?

Reply:
Yep great suggestion